src/mono/wasi/Wasi.Build.Tests/WasiTemplateTests.cs

```diff
@@ -81,40 +81,89 @@ public void ConsoleBuildThenPublish(string config, bool aot)
 
     [ConditionalTheory(typeof(BuildTestBase), nameof(IsUsingWorkloads))]
-    [MemberData(nameof(TestDataForConsolePublishAndRun))]
-    public void ConsolePublishAndRunForSingleFileBundle(string config, bool relinking, bool invariantTimezone)
+    [MemberData(nameof(TestDataForConsolePublishAndRunForSingleFileBundle), parameters: false)]
+    [MemberData(nameof(TestDataForConsolePublishAndRunForSingleFileBundle), parameters: true)]
+    public void ConsolePublishAndRunForSingleFileBundle_InvariantTimeZone(string config, bool invariantTimezone)
     {
+        string mainWithTzTest = """
+            using System;
+
+            Console.WriteLine("Hello, Wasi Console!");
+            for (int i = 0; i < args.Length; i ++)
+                Console.WriteLine($"args[{i}] = {args[i]}");
+
+            try
+            {
+                TimeZoneInfo tst = TimeZoneInfo.FindSystemTimeZoneById("Asia/Tokyo");
+                Console.WriteLine($"{tst.DisplayName} BaseUtcOffset is {tst.BaseUtcOffset}");
+            }
+            catch (TimeZoneNotFoundException tznfe)
+            {
+                Console.WriteLine($"Could not find Asia/Tokyo: {tznfe.Message}");
+            }
+
+            return 42;
+            """;
+
+        string extraProperties = invariantTimezone ? "<InvariantTimezone>true</InvariantTimezone>" : "";
+        CommandResult res = ConsolePublishAndRunForSingleFileBundleInternal(config, mainWithTzTest, extraProperties: extraProperties);
+        if(invariantTimezone)
+            Assert.Contains("Could not find Asia/Tokyo", res.Output);
+        else
+            Assert.Contains("Asia/Tokyo BaseUtcOffset is 09:00:00", res.Output);
+    }
+
+    [ConditionalTheory(typeof(BuildTestBase), nameof(IsUsingWorkloads))]
+    [MemberData(nameof(TestDataForConsolePublishAndRunForSingleFileBundle), parameters: false)]
+    [MemberData(nameof(TestDataForConsolePublishAndRunForSingleFileBundle), parameters: true)]
+    public void ConsolePublishAndRunForSingleFileBundle_InvariantGlobalization(string config, bool invariantGlobalization)
+    {
+        string mainWithGlobalizationTest = """
+            using System;
+            using System.Globalization;
+
+            Console.WriteLine("Hello, Wasi Console!");
+            for (int i = 0; i < args.Length; i ++)
+                Console.WriteLine($"args[{i}] = {args[i]}");
+
+            Console.WriteLine($"Number: {int.Parse("1", CultureInfo.InvariantCulture)}");
+            return 42;
+            """;
+
+        string extraProperties = invariantGlobalization ? "<InvariantGlobalization>true</InvariantGlobalization>" : "";
+        CommandResult res = ConsolePublishAndRunForSingleFileBundleInternal(config, mainWithGlobalizationTest, extraProperties: extraProperties);
+        Assert.Contains("Number: 1", res.Output);
+    }
+
+    private CommandResult ConsolePublishAndRunForSingleFileBundleInternal(string config, string programContents, string extraProperties = "", bool aot = false)
```

Comment:
Is it intentional to keep `aot` always false?

Reply:
Only till your PR adds the aot tests.